### PR TITLE
[SPARK-30583][DOC] Document LIMIT Clause of SELECT statement in SQL Reference

### PR DESCRIPTION
**Pull request by dilipbiswal**
### What changes were proposed in this pull request?
Document LIMIT clause of SELECT statement in SQL Reference Guide.

### Why are the changes needed?
Currently Spark lacks documentation on the supported SQL constructs causing
confusion among users who sometimes have to look at the code to understand the
usage. This is aimed at addressing this issue.

### Does this PR introduce any user-facing change?
Yes. 

**Before:**
There was no documentation for this.

**After.**
Screen Shot 2020-01-20 at 1 37 28 AM
Screen Shot 2020-01-20 at 1 37 43 AM

### How was this patch tested?
Tested using jykyll build --serve

### CODE REVIEW

File: docs/sql-ref-syntax-qry-select-limit.md [hunk 1]
```diff
@@ -18,5 +18,71 @@ license: |
...
+### Syntax
+{% highlight sql %}
+LIMIT { ALL | integer_expression }
+{% endhighlight %}
+
+### Parameters
+<dl>
+  <dt><code><em>ALL</em></code></dt>
+  <dd>
+    If specified, the query returns all the rows. In otherwords, no limit is applied if this
```
Comment (srowen) [inline docs/sql-ref-syntax-qry-select-limit.md]: other words

**Review by srowen — CHANGES_REQUESTED** [review]

File: docs/sql-ref-syntax-qry-select-limit.md [hunk 2]
```diff
@@ -18,5 +18,71 @@ license: |
...
+  +-------+---+
+  |Anil B |18 |
+  |Jack N |16 |
+  |John A |18 |
+  |Mike A |25 |
+  |Shone S|16 |
+  |Zen Hui|25 |
+  +-------+---+
+
+-- A function expression as a input to limit.
```
Comment (huaxingao) [inline docs/sql-ref-syntax-qry-select-limit.md]: ```a input``` -> ```an input```?

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by dilipbiswal** [comment]
Thank you @srowen @huaxingao

**Pull request closed without merging**